notify/mattermost/mattermost.go

```diff
@@ -0,0 +1,271 @@
+	logger  *slog.Logger
+	client  *http.Client
+	retrier *notify.Retrier
+
+	postJSONFunc func(ctx context.Context, client *http.Client, url string, body io.Reader) (*http.Response, error)
+}
+
+// New returns a new Mattermost notifier.
+func New(c *config.MattermostConfig, t *template.Template, l *slog.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "mattermost", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Notifier{
+		conf:         c,
+		tmpl:         t,
+		logger:       l,
+		client:       client,
+		retrier:      &notify.Retrier{},
+		postJSONFunc: notify.PostJSON,
+	}, nil
+}
+
+// request is the request for sending a Mattermost notification.
+// https://developers.mattermost.com/integrate/webhooks/incoming/#parameters
+type request struct {
+	Text        string                     `json:"text"`
+	Channel     string                     `json:"channel,omitempty"`
+	Username    string                     `json:"username,omitempty"`
+	IconURL     string                     `json:"icon_url,omitempty"`
+	IconEmoji   string                     `json:"icon_emoji,omitempty"`
+	Attachments []attachment               `json:"attachments,omitempty"`
+	Type        string                     `json:"type,omitempty"`
+	Props       *config.MattermostProps    `json:"props,omitempty"`
+	Priority    *config.MattermostPriority `json:"priority,omitempty"`
+}
+
+// attachment is used to display a richly-formatted message block for compatibility with Slack.
+// https://developers.mattermost.com/integrate/reference/message-attachments/
+type attachment struct {
+	Fallback   string                   `json:"fallback,omitempty"`
+	Color      string                   `json:"color,omitempty"`
+	Pretext    string                   `json:"pretext,omitempty"`
+	Text       string                   `json:"text,omitempty"`
+	AuthorName string                   `json:"author_name,omitempty"`
+	AuthorLink string                   `json:"author_link,omitempty"`
+	AuthorIcon string                   `json:"author_icon,omitempty"`
+	Title      string                   `json:"title,omitempty"`
+	TitleLink  string                   `json:"title_link,omitempty"`
+	Fields     []config.MattermostField `json:"fields,omitempty"`
+	ThumbURL   string                   `json:"thumb_url,omitempty"`
+	Footer     string                   `json:"footer,omitempty"`
+	FooterIcon string                   `json:"footer_icon,omitempty"`
+	ImageURL   string                   `json:"image_url,omitempty"`
+}
+
+// Notify implements the Notifier interface.
+func (n *Notifier) Notify(ctx context.Context, alert ...*types.Alert) (bool, error) {
+	var (
+		err  error
+		url  string
+		data = notify.GetTemplateData(ctx, n.tmpl, alert, n.logger)
+	)
+
+	if n.conf.WebhookURL != nil {
+		url = n.conf.WebhookURL.String()
+	} else {
+		content, err := os.ReadFile(n.conf.WebhookURLFile)
```

Comment:
I did not check yet, but do other notify integrations behave the same?
Basically config overriding a file config, if not then the behaviour must be documented.